force-app/main/default/classes/ReceiptTriggerHandler.cls

```
@@ -0,0 +1,83 @@
+            }
+        }
+
+        List<Contact> doctorsList = [SELECT Id, AccountId, Account.Department__c, Payments_amount__c, Account.Doctor_payments_to_receive_bonus__c, (SELECT Id
+                                                                                                                                                    FROM Cases
+                                                                                                                                                    WHERE Id IN:listCasesIdsFromReceipts)
+                                     FROM Contact ];
+
+        Map<Id, Contact> doctorsMap = new Map<Id, Contact>();
+        List<Id> doctorsDepartmentsIdsList = new List<Id>();
+
+        for (Contact contact_i : doctorsList){
+            doctorsMap.put(contact_i.Id, contact_i);
+            doctorsDepartmentsIdsList.add(contact_i.AccountId);
+        }
+
+        List<AggregateResult> allDoctorsTreatmentsList = [SELECT SUM(Amount_of_payments__c)amount, ContactId ctId
+                                                          FROM Case 
+                                                          WHERE ContactId IN:doctorsMap.keySet() AND CreatedDate = THIS_MONTH AND CreatedDate = THIS_YEAR
+                                                          GROUP BY ContactId];
+
+        List<Contact> doctorIdsToUpdateList = new List<Contact>();
+        //3.1
+        for (AggregateResult result_i : allDoctorsTreatmentsList){
+            if (doctorsMap.containsKey((Id) result_i.get('ctId'))){
+                if (doctorsMap.get((Id) result_i.get('ctId')).Account.Doctor_payments_to_receive_bonus__c < (Decimal) result_i.get('amount')){
+                    doctorIdsToUpdateList.add(new Contact(Id = (Id) result_i.get('ctId'), Payments_amount__c = (Decimal) result_i.get('amount')));
+                }
+
+            }
+        }
+        List<Database.SaveResult> updateResults = Database.update (doctorIdsToUpdateList, false);
```

use Database.update with allOrNothing flag = true, unless you are 100% sure you need to ignore exceptions. In this case I don't think you need to ignore them